cmd/controller/app/controller.go

```diff
@@ -267,7 +266,7 @@ func startLeaderElection(ctx context.Context, opts *options.ControllerOptions, l
 	}
 
 	// Try and become the leader and start controller manager loops
-	leaderelection.RunOrDie(context.TODO(), leaderelection.LeaderElectionConfig{
+	leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
```

Comment:
This fixes an issue where cert-manager won't exit if it is not the leader and gets a stop signal/ctrl+c